source/row_selection_table_model.ts

```diff
@@ -304,13 +350,16 @@ export class RowSelector {
 
   private state: number;
   private selectedRows: ArrayTableModel;
+  private highlightedRow: number;
+  private previousRow: number;
+  private currentRow: number;
   private isShiftDown: boolean;
   private isCtrlDown: boolean;
   private isMouseDown: boolean;
   private isUpDown: boolean;
   private isDownDown: boolean;
   private isAdding: boolean;
-  private hilightedRow: number;
-  private previousRow: number;
-  private currentRow: number;
+  private transactionCount: number;
```

Comment:
Similarly, none of 362, 363, 364 is needed. The whole point of having an ```ArrayTableModel``` is that all the work is done for you already by that class.